### ISSUE TITLE
Unable to cancel checkout twice

### ISSUE THREAD

**Issue by alexlivenson** [Affirm/affirm-ios-sdk#6]
There seems to be an issue with trying to cancel out the affirm checkout controller twice.
Step to reproduce.
1. Clone the example project https://github.com/Affirm/affirm-ios-sdk-demo
2. Select "Buy with Affirm" -> this will present the affirm web view
3. Tap the close icon on the left side and click "ok" on the pop up -> this will close the controller
4. **BUG:  Repeat steps 2 and 3 and you will notice that the affirm web view will not call the delegate**

I was able to reproduce this on xcode 8, ios 9 - 10

**Comment by affirmPE** [comment]
------------------------------------------------------------------------------------------------------------
Type your response on top of this email to respond to this case. Learn Email Commands Here - https://support.desk.com/customer/portal/articles/1191712-reply-via-email-commands
------------------------------------------------------------------------------------------------------------


Case Assignment Notification - Case #15827 ( https://affirm.desk.com/web/agent/case/15827 )

Affirm | Jun 12, 2017 09:51PM UTC

------------------------------------------------------------------------------------------------------------
Case #15827
Customer: Affirm/affirm-ios-sdk
Subject: [Affirm/affirm-ios-sdk] Unable to cancel checkout twice (#6)
------------------------------------------------------------------------------------------------------------
Case Information (15827):

Subject: [Affirm/affirm-ios-sdk] Unable to cancel checkout twice (#6)

Description: 
Status: Open
Priority: 4

Assigned Group: MTS



------------------------------------------------------------------------------------------------------------
Case History
------------------------------------------------------------------------------------------------------------
 goatmad  at JUN 12, 2017 | 02:51PM PDT:
There seems to be an issue with trying to cancel out the affirm checkout controller twice.
Step to reproduce.
1. Clone the example project https://github.com/Affirm/affirm-ios-sdk-demo
2. Select "Buy with Affirm" -> this will present the affirm web view
3. Tap the close icon on the left side and click "ok" on the pop up -> this will close the controller
**4. BUG:  Repeat steps 2 and 3 and you will notice that the affirm web view will not call the delegate**

I was able to reproduce this on xcode 8, ios 9 - 10

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/Affirm/affirm-ios-sdk/issues/6

------------------------------------------------------------------------------------------------------------
This message was sent to reply@reply.github.com in reference to Case #15827
------------------------------------------------------------------------------------------------------------

**Comment by affirmPE** [comment]
__________________________________
Type your response ABOVE THIS LINE to reply

------------------------------------------------------------------------------------------------------------
Customer: Affirm - V2
Subject: [Affirm/affirm-ios-sdk] Unable to cancel checkout twice (#6)
------------------------------------------------------------------------------------------------------------

Dipti | JUN 12, 2017 | 03:41PM PDT

Thanks for sending the details. We were able to reproduce this issue on our end. We will update you as soon as our developers are able to resolve this issue. Thank you for your patience.

--
Dipti Bele | Partner Engineer @ Affirm
Check out our new Merchant Help Site: docs.affirm.com
merchanthelp@affirm.com
------------------------------------------------------------------------------------------------------------

goatmad | JUN 12, 2017 | 02:51PM PDT

There seems to be an issue with trying to cancel out the affirm checkout controller twice.
Step to reproduce.
1. Clone the example project https://github.com/Affirm/affirm-ios-sdk-demo
2. Select "Buy with Affirm" -> this will present the affirm web view
3. Tap the close icon on the left side and click "ok" on the pop up -> this will close the controller
**4. BUG:  Repeat steps 2 and 3 and you will notice that the affirm web view will not call the delegate**

I was able to reproduce this on xcode 8, ios 9 - 10

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/Affirm/affirm-ios-sdk/issues/6



------------------------------------------------------------------------------------------------------------
This message was sent to reply@reply.github.com in reference to Case #: 15827.
------------------------------------------------------------------------------------------------------------




[[2bbf20d00b55d11accd8d7089a798554450d4db0-972573445]]
